src/main/kotlin/com/terraformation/backend/accelerator/ParticipantProjectSpeciesService.kt

```diff
@@ -190,11 +195,38 @@ class ParticipantProjectSpeciesService(
         FileMetadata.of(MediaType.valueOf("text/csv").toString(), filename, stream.size().toLong())
 
     fileService.storeFile("species-list-deliverable", inputStream, metadata, null) { fileId ->
-      submissionSnapshotsDao.insert(
-          SubmissionSnapshotsRow(submissionId = event.submissionId, fileId = fileId))
+      with(SUBMISSION_SNAPSHOTS) {
+        dslContext
+            .insertInto(this)
+            .set(SUBMISSION_ID, event.submissionId)
+            .set(FILE_ID, fileId)
+            .execute()
+      }
     }
   }
 
+  fun readSubmissionSnapshotFile(
+      projectId: ProjectId,
+      deliverableId: DeliverableId
+  ): SizedInputStream {
+    val submissionRecord =
+        with(SUBMISSIONS) {
+          dslContext.fetchOne(this, PROJECT_ID.eq(projectId).and(DELIVERABLE_ID.eq(deliverableId)))
+              ?: throw NotFoundException()
```

Comment:
Defining a more precise exception here would be good. It's not a huge deal, but `NotFoundException()` is from the web service library (where the exceptions and annotations for controllers live) and ideally, controller-specific stuff should be isolated in the `api` packages just to maintain more consistent layering. Plus, of course, a custom exception class can hold the two IDs for easier troubleshooting.